static/js/latest-news.js

```diff
@@ -0,0 +1,136 @@
+(function() {
+  function _slugify(string) {
+    var a =
+      "àáäâãåăæąçćčđďèéěėëêęğǵḧìíïîįłḿǹńňñòóöôœøṕŕřßşśšșťțùúüûǘůűūųẃẍÿýźžż·/_,:;";
+    var b =
+      "aaaaaaaaacccddeeeeeeegghiiiiilmnnnnooooooprrsssssttuuuuuuuuuwxyyzzz------";
+    var p = new RegExp(a.split("").join("|"), "g");
+    return string
```

Comment:
Whatever you're doing here looks like it should have been handled server-side by the `latest-news` endpoint. What is it trying to handle?

Reply:
This whole file was lifted directly from https://github.com/canonical-web-and-design/ubuntu.com/blob/master/static/js/latest-news.js

Reply:
Damnit! OK fair enough, happy to treat this as out-of-scope. But I still want an answer to the question at some point =)

Reply:
Wasn't it you that wrote it in the first place, or was that lifted from somewhere else too? 
https://github.com/canonical-web-and-design/ubuntu.com/commit/03298d6a7f32e59b6098f36100a249da69c4892e#diff-c3b4c9f4e86607ab79c6fb795d9bf13a

Reply:
Yeah I probably copied it from cn.ubuntu.com or snapcraft.io or something.

But also, just 'cos I wrote it doesn't mean a) it can't be terrible and b) that I might not have a strong objection to how I previously implemented it. I reserve the right to rail against my own code ;)

Reply:
Actually, I think I did author it, and I copied that function from here:

https://medium.com/@mhagemann/the-ultimate-way-to-slugify-a-url-string-in-javascript-b8e4a0d849e1

So there you go, entirely me. I still disapprove.

Reply:
Filed https://github.com/canonical-web-and-design/canonicalwebteam.blog/issues/130

Reply:
I agree you have every right to complain about your own code :)

I agree it's very hard to reason about, and looking at the file I'm not sure it's needed as `article.slug` seems to be thing: https://github.com/canonical-web-and-design/maas.io/blob/b3e84a7357131d875e007e86ea313253e2df1c11/static/js/latest-news.js#L46